### ISSUE TITLE
Invisible mode hides users from "Browse Members"... Members shows under "Featured Members"

### ISSUE THREAD

**Issue by hamadax** [PHPfox-Official/phpfox-v4-issues#2696]
# Important

Please follow this template!

----------------------------------------

## What's happened?
When a member has invisible mode turned on and is featured, he/she is hidden from the "browse members" section. By definition, invisible mode "Invisible mode allows you to browse the site without appearing on any "Online" lists."

Regardless whether the member is online or not, their profile does not show under browse members. On the other hand, they show up under featured members.

## Steps to reproduce:
1. Enable invisible mode on a user and feature him.
2. Browse members.
3. member shows up in featured, but not under browse members!

## What's expected?
...

## Browsers and Devices tested
(Example: Chrome on iPhone X, Safari on Macbook, Miscrosoft Edge on Windown 10, Firefox on Ubuntu 16.04, ...)

...

## Server information
(Example: CentOS 7, php 7.1 apache)

...

## phpFox version
(Example: phpFox 4.6.0)

...

## Screenshots
...

**Comment by harrison05** [comment]
I don't think this is a bug, it said "Invisible mode allows you to browse the site without appearing on any "Online" lists.", so when enable setting, users hide from "Recently Active" and "Friends Online" section but Featured Members still show because it doesn't show online status of them.

Regards.

**Comment by hamadax** [comment]
Well, if the user wants to stay invisible, then it is a blatant violation of the user's privacy to feature him/her in an online list of any kind! This mode "turned on" must disable / remove a member from any featured lists; if you follow your logic or the wording in place. It's clear bug to me... not sure why "featured members" are excluded from "online lists". IMO, it does not make common sense. "Online" can either be a user's online status or a user showing up in a list online. This feature is a mixture of both, since it hides their online status and removes them from the member directory, except "featured", thus it's an oxymoron and I cannot fix it with replacing the language phrase with an asterisk letting members know of an exception.

Also, whoever is using this feature is most likely going to have their profile settings set to private, so why feature a member that is not going to have anything publicly visible in their profile view!!

**Comment by hamadax** [comment]
The solution here is to not allow any member with this feature turned on to be featured. If they are featured already and they turn on this feature, then we hide them. Plain and simple.

**Comment by harrison05** [comment]
Your flow is not the business rules of this feature. 

>**Invisible mode allows you to browse the site without appearing on any "Online" lists..**

Turn on invisible mode doesn't mean absolutely hide that user from the site. It just hides the online state of them. Users still show on Feature block or their items show on feed stream, but no one knows they're online or not.

**Comment by hamadax** [comment]
OK, I will try to explain this again differently... There is a fundamental difference between "online status" and "online lists". Invisible mode in PHPFox is a mixture of both, which is wrong by design and needs to be split. IMO, whoever coded this feature decided to combine two different features into one with a phrase describing only one feature and no way to rephrase to describe both.

Invisible mode as it should is like your Skype status: Active, Away, Invisible etc... On the other hand, a profile setting that hides the user profile from online directories/lists/searches has nothing to do with an online status. Here is how it's done in Skype, there are two separate settings:


![image](https://user-images.githubusercontent.com/48427415/60190873-5f228d80-9801-11e9-82e8-861aa7012b26.png)

![image](https://user-images.githubusercontent.com/48427415/60190898-6ba6e600-9801-11e9-82b9-9947ad8a3605.png)

Let's see: Invisible mode in PHPFox does the following:

- Hides the online status of the user. He appears to be offline. We are telling the user they will not show in "online lists". Nothing about their "online status".
- When turned on, it does not remove a member from "Recently Active" right away. They need to log out and log back in! Sometimes, it takes time to take effect. Not really sure what's happening here.
- With invisible mode turned on, user still shows up in friends suggestions. Not really invisible. Isn't that an online list of suggested people!
- Hides the user profile from the member directory. This is what we are telling the user, but they still show up as featured. Featured is an online list of featured members.
- While invisible, they can bypass the blog view counter and their views count with every single page refresh. Admin setting has no effect on blog page views when the user is in invisible mode.

Invisible mode in any online app has nothing to do with online lists or showing up in online directories or searches. That needs to be a different setting under the profile on the user. Granular by section. Take any social app in the market and see how they separate status and appearing in lists/searches.

This has nothing to do with business models.

If you were to actually put an English phrase that accurately describes this feature, what would it be so we let that user know what it does exactly?!
